tests/components/test_rest_command.py

```diff
@@ -240,3 +240,22 @@ def test_rest_command_content_type(self, aioclient_mock):
 
         assert len(aioclient_mock.mock_calls) == 1
         assert aioclient_mock.mock_calls[0][2] == b'item'
+
+    def test_rest_command_headers(self, aioclient_mock):
+        """Call a rest command with custom headers."""
+        data = {
+            'payload': 'item',
+            'headers': {'Accept': 'application/json'}
```

Comment:
Am I correct in saying that this only really tests config validation?

Maybe combine it in one method, only running up to `assert_setup_component` and then test all combinations:

* Nothing
* content-type
* headers
* headers & content type

Reply:
Indeed, though that's true of all the existing tests in that file. I simply copied and modified the existing test for the content-type. While the tests could all be improved in that file, the changes to the tests would be relatively extensive which rather expands the scope of this relatively minor change. I'm also not familiar with home-assistant's testing framework (yet).

Reply:
For now, just fix yor new test.

Just remove everything after the `assert_setup_component` block and add the various options & assert blocks. You might have to clear existing components between the tests something like `hass.components=[]`

See [here](https://github.com/home-assistant/home-assistant/blob/dev/tests/common.py#L521) why/how it tests validation

Use `pytest -k rest` for quick tests

Reply:
Thanks for the tips. I've updated the test to check the various combinations of header modifying parameters have the intended effect on the headers in the request.